GetDKAN/dkan
# CSV downloads for Datastore

This PR adds the options to get retrieve datastore data as a CSV. This is done by adding a property "format" to the query, and setting the value to "csv" (in either the query string of a GET request or the body of a POST). Supplying nothing or format=json will return json as before. The first row of the file will be column names from the schema/fields portion of the query results. 

Additionally, this adds a series of new endpoints with the addition of /download to the path. Currently this only works on CSV requests, and rather than limiting the results to 500 rows will stream the full result to a file for download. This endpoint may have implications for load/performance and should be used with caution; additional permissions to block access by default may be added in the future.

Also included in this PR is an update to the open_api spec. There has been for some time an undocumented datastore endpoint that includes the distribution/resource UUID as a path parameter, and rejects adding a resources array to the query object. It would be useful to publish a second schema for this endpoint that is more limited, as the full datastore query method is not actually supported (in addition to not allowing resources in the query, joins are not possible either).

## QA Steps
- Find the UUID of a distribution in DKAN
- Request /api/1/datastore/query/{identifier}?format=json and verify valid json is still returned. Add ?format=csv to a second request and verify a csv file is returned, and that it has 500 rows.
- Make a request to /api/1/datastore/query/{identifier}/download?format=json and verify a csv file is returned. Confirm you have received the full result set.
